apps/captain_fact/lib/captain_fact/authenticator/authenticator.ex

```diff
@@ -10,11 +11,16 @@ defmodule CaptainFact.Authenticator do
   alias Kaur.Result
 
   @doc """
-  Get user from its email address and check password.
+  Get user from its email address or user name and check password.
   Returns nil if no User for email or if password is invalid.
   """
-  def get_user_for_email_password(email, password) do
-    with user when not is_nil(user) <- Repo.get_by(User, email: email),
+  def get_user_for_email_or_name_password(email_or_name, password) do
+    user =
+      User
+      |> where([u], u.email == ^email_or_name or u.name == ^email_or_name)
```

Comment:
`User` schema fields may be confusing. `name` is meant to store User's real name and is optional. The field you're looking for is **u.username** which is not nullable and unique among users.

Reply:
Ooops good one. Thanks! Fixed :)